js/jquery.imageplayer.js

```diff
@@ -169,21 +170,17 @@ if (typeof(jQuery) == 'undefined') alert('jQuery library was not found.');
         
         function image_transition(img) {
             set_image(img);
-            // animate scrubber
-            last_frame_scrubber_pos = parseFloat(scrubber_handle.css('left'));
-            var remaining = inc*(i+1) - last_frame_scrubber_pos;
-            // var percent = Math.floor(remaining / inc);
-            scrubber_handle.stop(true, true);
-            scrubber_handle.animate({
-                left: '+='+remaining+'px'
-            }, settings.delay*1000, 'linear');
+            handle = document.getElementById('range');
+            //animate scrubber 
+            if( i != 0 ){
+              handle.stepUp(1);
```

Comment:
As a global idea, and now that we switched to an input range slider:

Why not just trigger programmatically the "change" event on the input slider (every time we modify its value) and make the change listener set the image?

In other words, every time we change the value programatically, trigger the event:

```js
scrubber.val(new_value);
scrubber.trigger('change'); // this would call the function bind to the change event, that function would set the image
```

Does this make sense and make the flow clearer for you too?